controlplane/pkg/nsmd/network_service_server.go

```diff
@@ -102,7 +102,7 @@ func (srv *networkServiceServer) Request(ctx context.Context, request *networkse
 		defer dataplaneConn.Close()
 	}
 
-	endpoint, err := srv.model.SelectEndpoint(request.GetConnection().GetNetworkService())
+	endpoint, err := srv.getEndpoint(request)
```

Comment:
Could we preserve the round robin behavior of the SelectEndpoint?  Its important for the integration testing.

Reply:
Ah, I see, you do unless there's an error... got it :)

Reply:
Yes, it ask for local ones and if there is none go for nsmd-k8s registry.